### PR TITLE
Simplify gitlab ci

### PR DESCRIPTION
**Pull request by white238**
* removes dead devtools code
* simplifies stages
* renames workflows to full or src
* unifies blueos and toss3 scripts into one
* removes duplicate building of src in full build (build_tpl build src after building tpls so you don't have to call build_src afterwards)
* makes full build directory unique

### CODE REVIEW

**Comment by white238** [comment]
LGTM

**Comment by white238** [comment]
LGTM

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -7,120 +7,56 @@ variables:
...
-# There are no tests for now
 stages:
-  - l_build
-  - l_build_with_deps
-  - q_allocate_resources
-  - q_build
-  - q_build_with_deps
-  - q_build_devtools
-  - q_release_resources
+  - allocate
```
Comment (jamiebramwell) [inline .gitlab-ci.yml:13]: I really like these simplified stages. It makes the DAG much easier to read.

**Review by jamiebramwell — APPROVED** [review]

**Review by joshessman-llnl — APPROVED** [review]

**Comment by white238** [comment]
LGTM

**Pull request merged**